src/proxy/http/canonicalize.rs

```diff
@@ -179,33 +188,62 @@ impl Future for Task {
 
     fn poll(&mut self) -> Poll<(), ()> {
```

Comment:
Just curious why this is modeled as a Future instead of as a Stream?

Reply:
You can only spawn futures. As this is intended to be a spawned background task, it's a future that only completes when all work is done.

Or, to put it another way: what would consume the stream?

Reply:
I'd imagine the service would consume it.  i.e. rather than doing `self.rx.poll()` doing `self.addr_stream.poll()`.

Reply:
But I guess the difference is that this way the DNS resolutions are refreshed eagerly.

Reply:
Exactly. We rarely want to use poll_ready to drive work; almost always it should just check if other work has completed.